back/engines/commercial/insights/spec/services/insights/category_assignments_service_spec.rb

```diff
@@ -43,8 +43,9 @@
       categories = create_list(:category, 3)
 
       aggregate_failures 'check categories input_count' do
-        service.add_assignments(input, categories[1..2])
+        service.add_assignments(input, categories.drop(1))
 
+        categories.each(&:reload)
```

Comment:
The test was failing because the categories were not reloaded, so their input count was not up to date. The other changes are mere simplifications.